CMakeLists.txt

```diff
@@ -124,12 +127,12 @@ if(NOT SPDLOG_FMT_EXTERNAL AND NOT SPDLOG_FMT_EXTERNAL_HO)
     list(APPEND SPDLOG_SRCS src/fmt.cpp)
 endif()
 
-if(SPDLOG_BUILD_SHARED)
-    if(WIN32)
-        configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/version.rc.in ${CMAKE_CURRENT_BINARY_DIR}/version.rc @ONLY)
-        list(APPEND SPDLOG_SRCS ${CMAKE_CURRENT_BINARY_DIR}/version.rc)
-    endif()
-    add_library(spdlog SHARED ${SPDLOG_SRCS} ${SPDLOG_ALL_HEADERS})
+if(BUILD_SHARED_LIBS AND WIN32)
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/version.rc.in ${CMAKE_CURRENT_BINARY_DIR}/version.rc @ONLY)
+    list(APPEND SPDLOG_SRCS ${CMAKE_CURRENT_BINARY_DIR}/version.rc)
+endif()
+add_library(spdlog ${SPDLOG_SRCS} ${SPDLOG_ALL_HEADERS})
+if (BUILD_SHARED_LIBS)
```

Comment:
No need for any changes from orig code except for the ```if(SPDLOG_BUILD_SHARED OR BUILD_SHARED_LIBS) ```

Reply:
Your proposition will not work. If I set `SPDLOG_BUILD_SHARED` to true w/o setting `BUILD_SHARED_LIBS` to true, `add_library` will still build a static library.